prometheus/src/cluster_metrics.rs

```diff
@@ -1,12 +1,84 @@
 use solana_gossip::cluster_info::ClusterInfo;
 use solana_runtime::bank::Bank;
+use solana_sdk::{clock::Slot, pubkey::Pubkey};
+use solana_vote_program::vote_state::VoteState;
 
 use crate::{
     token::Lamports,
     utils::{write_metric, Metric, MetricFamily},
 };
 use std::{io, sync::Arc, time::SystemTime};
 
+struct ValidatorVoteInfo {
+    vote_address: Pubkey,
+    balance: Lamports,
+    last_vote: Slot,
+}
+
+impl ValidatorVoteInfo {
+    fn new_from_bank(bank: &Arc<Bank>, identity_pubkey: &Pubkey) -> Option<Self> {
+        let vote_accounts = bank.vote_accounts();
+        let vote_state_default = VoteState::default();
+        vote_accounts
+            .iter()
+            .filter_map(|(&vote_pubkey, (_activated_stake, account))| {
+                let vote_state = account.vote_state();
+                let vote_state = vote_state.as_ref().unwrap_or(&vote_state_default);
+                if identity_pubkey != &vote_state.node_pubkey {
+                    return None;
+                }
+                let last_vote = if let Some(vote) = vote_state.votes.iter().last() {
+                    vote.slot
+                } else {
+                    0
```

Comment:
Instead of returning 0, we could make the metric itself absent if there is no last vote. That is more accurate, but I think in practice it will also be nicer to work with, if you run a `delta` or `rate` query on this metric, if it jumps to 0 for some reason you get enormous deltas. If the metric is absent, then the rate/delta is just not defined and absent itself.